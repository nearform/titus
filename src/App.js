import React, { Component } from 'react';
import MainNav from './nav/MainNav';
import BasicContent from './content/BasicContent';
import CssBaseline from '@material-ui/core/CssBaseline';
import { MuiThemeProvider } from '@material-ui/core/styles';
import { theme } from './theme/theme';
import { metaData } from './mock/appData';

class App extends Component {
  render() {
    return (
      <React.Fragment>
        <CssBaseline />
        <MuiThemeProvider theme={theme}>
          <MainNav appName={metaData.name}>
            <BasicContent textContent={metaData.description} />
          </MainNav>
        </MuiThemeProvider>
      </React.Fragment>
    );
  }
}

export default App;
