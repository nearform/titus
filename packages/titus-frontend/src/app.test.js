import React from 'react'
import { render, cleanup } from '@testing-library/react'

import App from './app'

describe('<App />', () => {
  afterEach(cleanup)

  it('should render correctly when path is / and is not authenticated', () => {
    jest.mock('./history', () => ({
      length: 1,
      location: {
        pathname: '/'
      },
      listen: jest.fn()
    }))

    const { asFragment } = render(<App />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly when path is /login and is not authenticated', () => {
    jest.mock('./history', () => ({
      length: 1,
      location: {
        pathname: '/login'
      },
      listen: jest.fn()
    }))

    const { asFragment } = render(<App />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly when path is /login and is authenticated', () => {
    jest.mock('./history', () => ({
      length: 1,
      location: {
        pathname: '/login'
      },
      listen: jest.fn()
    }))

    const { asFragment } = render(<App />)

    expect(asFragment()).toMatchSnapshot()
  })
})
