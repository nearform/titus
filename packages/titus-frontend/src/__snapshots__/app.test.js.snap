// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly when path is / and is not authenticated 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="loader"
    />
  </div>
</DocumentFragment>
`;

exports[`<App /> should render correctly when path is /login and is authenticated 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <svg
      alt="Titus logo"
      class="login-logo"
    >
      Accel_Logo_Titus.svg
    </svg>
    <form
      novalidate=""
    >
      <header>
        Note: Any username followed by a password with at least four characters containing at least one letter and number will work.
      </header>
      <input
        aria-label="Username:"
        id="username"
        name="username"
        placeholder="Username"
        required=""
        type="text"
        value=""
      />
      <input
        aria-label="Password:"
        id="password"
        name="password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <button
        class="button"
        type="submit"
      >
        Login
      </button>
      <sub />
    </form>
  </div>
</DocumentFragment>
`;

exports[`<App /> should render correctly when path is /login and is not authenticated 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <svg
      alt="Titus logo"
      class="login-logo"
    >
      Accel_Logo_Titus.svg
    </svg>
    <form
      novalidate=""
    >
      <header>
        Note: Any username followed by a password with at least four characters containing at least one letter and number will work.
      </header>
      <input
        aria-label="Username:"
        id="username"
        name="username"
        placeholder="Username"
        required=""
        type="text"
        value=""
      />
      <input
        aria-label="Password:"
        id="password"
        name="password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <button
        class="button"
        type="submit"
      >
        Login
      </button>
      <sub />
    </form>
  </div>
</DocumentFragment>
`;
